### PR TITLE
Add Outline image effect to post processing effects system

### PR DESCRIPTION
**Pull request by herocrab**
# PR Details

Outline PostEffect

## Description

This change provides cartoon-like outline image effect to the post processing effects system. The content was mined from phr00t's Focus fork and given small clean-up. I am not the original author of the shader. It appears as though phr00t's fork is open source following the same license as Stride.

outline-effect-1

I have not written any new tests for this feature.

I submitted a separate PR for documentation update [here](https://github.com/stride3d/stride-docs/pull/63).

## Motivation and Context

This item has been requested multiple times in Discord and is a popular effect for games.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by herocrab** [comment]
Documentation is here: https://github.com/stride3d/stride-docs/pull/63

File: sources/engine/Stride.Rendering/Rendering/Images/Outline/Outline.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Xenko contributors (https://xenko.com) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
```
Comment (xen2) [inline sources/engine/Stride.Rendering/Rendering/Images/Outline/Outline.cs:1]: Please change to `// Copyright (c) Stride contributors (https://stride3d.net)`

File: sources/engine/Stride.Rendering/Rendering/Images/Outline/OutlineEffect.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// <auto-generated>
+// Do not edit this file yourself!
+//
+// This code was generated by Xenko Shader Mixin Code Generator.
+// To generate it yourself, please install Xenko.VisualStudio.Package .vsix
+// and re-save the associated .xkfx.
```
Comment (xen2) [inline sources/engine/Stride.Rendering/Rendering/Images/Outline/OutlineEffect.cs:6]: Same as other PR (`Xenko` -> `Stride` and `.xkfx` -> `.sdfx`)

File: sources/engine/Stride.Rendering/Rendering/Images/PostProcessingEffects.cs [hunk 3]
```diff
@@ -39,6 +39,7 @@ public PostProcessingEffects(IServiceRegistry services)
         /// </summary>
         public PostProcessingEffects()
         {
+            Outline = new Outline {Enabled = false};
```
Comment (xen2) [inline sources/engine/Stride.Rendering/Rendering/Images/PostProcessingEffects.cs:42]: curly bracket spacing

File: sources/engine/Stride.Rendering/Rendering/Images/Outline/Outline.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+using Stride.Core;
+using Stride.Core.Mathematics;
+using Stride.Graphics;
+
+namespace Stride.Rendering.Images {
+    /// <summary>
+    /// A fog filter.
+    /// </summary>
+    [DataContract("Outline")]
+    public class Outline : ImageEffect {
```
Comment (xen2) [inline sources/engine/Stride.Rendering/Rendering/Images/Outline/Outline.cs:14]: Curly bracket should be on their own line to match rest of codebase

**Comment by herocrab** [comment]
Thanks for the review. These are simple things I missed in the flurry, my apologies. Updated.

**Pull request merged**